static/usr/lib/core/rtc-sys-time-init

```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
+
+# Adjust system time for the following reasons:
+#     1) Sync system time to RTC time
+#          - On older kernel versions, modules with HTCOSYS enabled cannot sync system time with
+#            RTC (unlike built-in RTC drivers)
+#          - Where it can sync at driver level, this might cause harmless re-sync
+#     2) Correct driver/module system time sync to invalid time (earlier than most recent time stamp)
+#          - Built-in/loadable RTC driver with HCTOSYS enabled may set invalid time due to uninitialized,
+#            invalid time e.g. caused by bad battery
+#
+# If the RTC time is not valid, or cannot be set, forward the system time to most recent timestamp
+
+# Expects single argument that is service instance name which is
+# based on the associated RTC device unit name in the form dev-rtc*
+dev_unit="$1"
+
+# Get systemd timestamp file created during core build
+clock_epoch_stamp=0
+if [ -e "/usr/lib/clock-epoch" ]; then
+    clock_epoch_stamp="$(stat -L "/usr/lib/clock-epoch" -c '%Y')"
+fi
+clock_epoch_date=$(date -d @"$clock_epoch_stamp")
+echo "core/systemd build timetamp: $clock_epoch_date"
+
+# Get systemd-timesyncd timestamp file initialized to systemd build time and updated periodically
+# (default 60s) and at shutdown
+timesync_clock_stamp=0
+if [ -e "/var/lib/systemd/timesync/clock" ]; then
+    timesync_clock_stamp="$(stat -L "/var/lib/systemd/timesync/clock" -c '%Y')"
+fi
+timesync_clock_date=$(date -d @"$timesync_clock_stamp")
+echo "timesyncd timestamp:         $timesync_clock_date"
+
+most_recent_stamp=$(echo "$clock_epoch_stamp $timesync_clock_stamp" | tr ' ' '\n' | sort -rn | head -1)
+
+sys_time_set_res=0
+dev_node=$(echo "$dev_unit" | sed 's/dev-rtc/\/dev\/rtc/')
+echo "Reading RTC $dev_node time..."
+if ! rtc_stamp=$(hwclock -r -u -f "$dev_node" 2>/dev/null); then
+    sys_time_set_res=1
+    echo "Error: cannot read RTC $dev_node"
+else
+    rtc_stamp=$(date -d "$rtc_stamp" +%s)
+    rtc_date=$(date -d @"$rtc_stamp")
+    echo "Successfully read RTC $dev_node time: $rtc_date"
+
+    if [ "$rtc_stamp" -gt "$most_recent_stamp" ]; then
+        echo "RTC time is valid, syncing system time..."
+        if hwclock -s --noadjfile -u -f "$dev_node" 2>/dev/null; then
```

Comment:
This should happen only when `hctosys` is 0.

Reply:
Great suggestion, thanks. Done.